packages/wonder-blocks-dropdown/components/dropdown-core-virtualized.js

```diff
@@ -1,113 +1,206 @@
-                role,
-                style,
-            });
+/**
+ * A react-window's List wrapper that instantiates the virtualized list and
+ * dynamically calculates the item height depending on the type
+ */
+class DropdownCoreVirtualized extends React.Component<Props, State> {
+    state = {
+        height: this.getHeight(),
+        width: null,
+    };
+
+    componentDidMount() {
+        // Wait for styles to be applied. This way, we can get a more precise
+        // value of the container dimensions.
+        setTimeout(() => {
+            this.setWidth();
+        }, 0);
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        const {data, listRef} = this.props;
+
+        // if the items size has changed, then recalculate each item position
+        if (prevProps.data.length !== data.length) {
+            this.setHeight();
+
+            if (listRef && listRef.current) {
+                // the ref can't associate this instance method
+                // $FlowIgnore
+                listRef.current.resetAfterIndex(1);
+            }
         }
     }
-}
 
-type Props = {|
     /**
-     * The complete list of items that will be virtualized.
+     * Update container width
      */
-    data: Array<DropdownItem>,
+    setWidth() {
+        const rootNode = ((ReactDOM.findDOMNode(this): any): ?HTMLElement);
+
+        // after the non-virtualized items are rendered, we get the container
+        //  width to pass it to react-window's List
+        if (rootNode) {
+            this.setState({
+                width: rootNode.getBoundingClientRect().width,
+            });
+        }
+    }
 
     /**
-     * The list height that needs to be passed in order to let react-window
-     * calculate each item vertical position
+     * Update container height
      */
-    height: number,
+    setHeight() {
+        // calculate dropdown's height depending on the type of items
+        const height = this.getHeight();
+        this.setState({height});
+    }
 
     /**
-     * The list width that needs to be passed in order to let react-window
-     * calculate each item position
+     * The list height that is automatically calculated depending on the
+     * component's type of each item (e.g. Separator, Option, Search, etc)
      */
-    width: number,
-|};
+    getHeight() {
+        // calculate using the first 10 items on the array as we want to display
+        // this number of elements in the visible area
+        return this.props.data
+            .slice(0, MAX_VISIBLE_ITEMS)
+            .reduce((sum, item) => {
+                if (SeparatorItem.isClassOf(item.component)) {
+                    return sum + 9;
+                } else if (SearchTextInput.isClassOf(item.component)) {
+                    // search text input height
+                    return sum + SEARCH_ITEM_HEIGHT;
+                } else {
+                    return sum + DROPDOWN_ITEM_HEIGHT;
+                }
+            }, 5);
```

Comment:
Starting with 5? Can we make this a constant and document it?

Reply:
Oh good catch, I think it was me playing around when defining this function. I forgot to remove that :|